code/game/machinery/reagent_forge.dm

```diff
@@ -0,0 +1,112 @@
+/obj/machinery/reagent_forge
+	name = "material forge"
+	desc = "A bulky machine that can smelt practically any material in existence."
+	icon = 'icons/obj/3x3.dmi'
+	icon_state = "arc_forge"
+	bound_width = 96
+	bound_height = 96
+	anchored = TRUE
+	max_integrity = 1000
+	density = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 10
+	active_power_usage = 3000
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	circuit = null
+	light_range = 5
+	light_power = 1.5
+	light_color = LIGHT_COLOR_FIRE
+	var/operation = ""
+	var/datum/reagent/currently_forging //forge one mat at a time
+	var/processing = FALSE
+	var/efficiency = 1
+	var/datum/techweb/stored_research
+	var/list/currently_using = list()
+	var/trueamount = 0
+
+/obj/machinery/reagent_forge/examine(mob/user)
+	//go fuck yourself if you think I'm using a SIGNAL FOR FUCKING EXAMINING
+	. = ..()
+	if(trueamount != 0)
+		. += "\n The Forge contains [span_blue(trueamount)] units of [span_blue(currently_forging)]"
+	else
+		. += "\n The Forge is empty."
+
+/obj/machinery/reagent_forge/attackby(obj/item/stack/sheet/mineral/reagent/I, mob/living/carbon/human/user)
+	if(istype(I, /obj/item/stack/sheet/mineral/reagent))
+		var/obj/item/stack/sheet/mineral/reagent/R = I
+		if(R.reagent_type)
+			var/datum/reagent/RE = R.reagent_type
+			if(!initial(RE.can_forge))
+				to_chat(user, "<span class='warning'>[initial(RE.name)] cannot be forged!</span>")
+				return
+			if(!currently_forging || !currently_forging.type)
+				if(R.amount <= 0)//this shouldn't exist
```

Comment:
this is probably pretty pointless

Reply:
A check for adminbus, innit. Admins can vv the amount.

Reply:
if they vv they amount then an admin vv'd the amount, its usuall just annoying rather than helpful when the code is fucking with your attempts to screw with things

Reply:
Fair.